File: 프로그래머스/1/160586. 대충 만든 자판/대충 만든 자판.js
function solution(keymap, targets) {
  const answer = [];
  const dic = {};

  keymap.forEach((key) => {
    key.split('').forEach((k, idx) => {
      if (dic[k] !== undefined && idx > dic[k]) {
        return;
      }
      dic[k] = idx;
    });
  });

  for (let i=0; i<targets.length; i++){
      let result = 0;
      for (let j=0; j<targets[i].length; j++){
          if (dic[targets[i][j]] === undefined){
              result = -1;
              break;
          } else {
              result += dic[targets[i][j]] + 1;
          }
      }
      answer.push(result)
  }

  return answer;
}

File: 프로그래머스/1/160586. 대충 만든 자판/대충 만든 자판.py
def solution(keymap, targets):
    answer = []
    dic = dict()    
    
    for key in keymap:
        for idx, k in enumerate(key):
            if k in dic and idx > dic[k]:
                continue
            dic[k] = idx
            
    
    for tar in targets:
        tmp = 0
        for t in tar:
            if t not in dic:
                tmp = -1
                break
            else:
                tmp += (dic[t]+1)
        answer.append(tmp)
    return answer